stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/home/HomeFragment.kt

```diff
@@ -58,6 +67,66 @@ class HomeFragment : Fragment() {
     private lateinit var avatarView: AvatarView
     private lateinit var nameTextView: TextView
 
+    private val requestPermissionLauncher = registerForActivityResult(
+        ActivityResultContracts.RequestPermission()
+    ) { isGranted: Boolean ->
+        Log.i(TAG, "[onActivityResult] isGranted: $isGranted")
+        if (isGranted) {
+            // Permission is granted. Continue the action or workflow in your
+            // app.
+            sendNotification(this)
+        } else {
+            // Explain to the user that the feature is unavailable because the
+            // features requires a permission that the user has denied. At the
+            // same time, respect the user's decision. Don't link to system
+            // settings in an effort to convince the user to change their
+            // decision.
+            showNotificationBlocked()
+        }
+    }
+
+    private fun sendNotification(fragment: Fragment) {
+        Toast.makeText(fragment.requireContext(), "This is notification!", Toast.LENGTH_SHORT).show()
+    }
+
+    private fun onRequestNotificationClick() {
+        when {
+            ContextCompat.checkSelfPermission(
+                requireContext(), Manifest.permission.POST_NOTIFICATIONS
+            ) == PackageManager.PERMISSION_GRANTED -> {
+                // You can use the API that requires the permission.
+                Log.i(TAG, "[onRequestNotificationClick] PERMISSION GRANTED")
+                sendNotification(this)
+            }
+            shouldShowRequestPermissionRationale(Manifest.permission.POST_NOTIFICATIONS) -> {
+                Log.i(TAG, "[onRequestNotificationClick] PERMISSION BLOCKED")
+                showNotificationBlocked()
+            }
+            else -> {
+                Log.i(TAG, "[onRequestNotificationClick] PERMISSION REQUESTED")
+                // The registered ActivityResultCallback gets the result of this request
+                requestPermissionLauncher.launch(
+                    Manifest.permission.POST_NOTIFICATIONS
+                )
+            }
+        }
+    }
+
+    private fun showNotificationBlocked() {
+        Snackbar.make(
+            requireView(),
+            "Notification blocked",
+            Snackbar.LENGTH_LONG
+        ).setAction("Settings") {
+            // Responds to click on the action
+            val intent = Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS)
+            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            val uri: Uri = Uri.fromParts("package", requireContext().packageName, null)
+            intent.data = uri
+            startActivity(intent)
+        }.show()
+    }
+
```

Comment:
Is this code needed anymore?
Was it debug code to perform some tests?

Reply:
Yeah, that's for DEBUG purposes only